### DISCUSSION TITLE
Warning BE0066 ResolveReferenceLinksComponent

### DISCUSSION THREAD

**Discussion by nblackburn77** [EWSoftware/SHFB#1025] [Q&A] [answered]
We include a third party (CefSharp) documentation file in our help as we extend one of their controls.

On building I get several 'Unknown reference link' target warnings because the CefSharp documentation is referencing one of their private methods, OnAfterBrowserCreated:

`SHFB: Warning BE0066: ResolveReferenceLinksComponent: [T:CefSharp.WinForms.ChromiumWebBrowser] Unknown reference link target 'M:CefSharp.WinForms.ChromiumWebBrowser.OnAfterBrowserCreated(CefSharp.IBrowser)'.`

It also throws the same warning in our extended component:

`SHFB: Warning BE0066: ResolveReferenceLinksComponent: [T:Company.WebBrowser] Unknown reference link target 'M:CefSharp.WinForms.ChromiumWebBrowser.OnAfterBrowserCreated(CefSharp.IBrowser)'.`

I've tried putting all third party sources in a separate help file and linking to it using the Additional Reference Links plug-in and setting all links to none, but the warnings remain.

I can't use the API filter to exclude Company.WebBrowser and CefSharp.WinForms.ChromiumWebBrowser as this creates lots of unknown reference link targets from other areas of our API.

We don't include private members in our help.

Is there any other way around this that I maybe missing?

Thanks.

**Comment by EWSoftware** [comment]
Are you including the private members in the linked project?  If not, it won't be there to ignore.

**Reply by nblackburn77** [reply]
I wasn't, turning on private members in the linked project has removed these warnings:

`SHFB: Warning BE0066: ResolveReferenceLinksComponent: [T:CefSharp.WinForms.ChromiumWebBrowser] Unknown reference link target 'M:CefSharp.WinForms.ChromiumWebBrowser.OnAfterBrowserCreated(CefSharp.IBrowser)'.`

But I'm still getting the warnings from our type that extends ChromiumWebBrowser:

```
SHFB: Warning BE0066: ResolveReferenceLinksComponent: [T:Company.WebBrowser] Unknown reference link target 'M:CefSharp.WinForms.ChromiumWebBrowser.OnAfterBrowserCreated(CefSharp.IBrowser)'.
SHFB: Warning BE0066: ResolveReferenceLinksComponent: [Methods.T:Company.WebBrowser] Unknown reference link target 'M:CefSharp.WinForms.ChromiumWebBrowser.OnAfterBrowserCreated(CefSharp.IBrowser)'.
```
Thanks.

**Reply by nblackburn77** [reply]
Do you think the only solution is to also turn on private members in our main help project? That's the only way it'll see OnAfterBrowserCreated I guess?

**Reply by EWSoftware** [reply, marked as answer]
No, you'd end up with a lot of unwanted members you'd have to remove with the API filter.  More likely than not, it's referenced in a public member's comments.  If you can track it down, you could create a local XML comments file with a copy of the comments for that member with the link removed.  It would override the inherited comments.

**Reply by nblackburn77** [reply]
There was no reference to it in our code, but I've overridden the method that references the private member in our class, and copied the documentation with the link removed as suggested. This has fixed it.

Many thanks for your help.
